linode/manager
# upcoming: [M3-8465] – Add "Volume Encryption" section to "Create and Attach Volume" drawer

## Description 📝
Add "Volume Encryption" section to "Create and Attach Volume" drawer

## Changes  🔄
- Add `useVolumeQuery`
  - (tangential refactor for `useDetachVolumeMutation` to accommodate invalidation of `useVolumeQuery`)
- `VolumeSelect` refactor to use our custom `Autocomplete` component rather than MUI's directly
- Conditionally-displayed reboot notice in `LinodeVolumeAddDrawer`
  - E2E test coverage for this
- "Volume Encryption" section added in `LinodeVolumeCreateForm`
  - Unit test coverage for this

## Target release date 🗓️
9/16/24

## Preview 📷
| Before  | After |
| ------------- | ------------- |
| ![Screenshot 2024-08-29 at 4 30 47 PM](https://github.com/user-attachments/assets/609e2796-50be-487e-88f0-e7f09fcbbee2)  | ![Screenshot 2024-08-29 at 4 31 17 PM](https://github.com/user-attachments/assets/2a9da0c3-b7c3-4f5e-a621-9e6fd69416e1)  |
| ![Screenshot 2024-08-29 at 4 31 45 PM](https://github.com/user-attachments/assets/99b0c987-8411-43b0-aca9-3af339eff3d9)  | ![Screenshot 2024-08-29 at 4 32 06 PM](https://github.com/user-attachments/assets/7584d7ac-faa2-445b-8b15-71a233515043)  |

## How to test 🧪
### Prerequisites
Point at the dev environment with the `blockstorage-encryption` tag on your account. The linode you use should have a `capabilities` array that does not contain `blockstorage_encryption`. Without one or both of those, you should not observe any of the below.

### Verification steps
- "Create and Attach Volume" view
  - check the "Encrypt Volume" checkbox --> observe: reboot notice below radio buttons & caveats in notice above checkbox, disabled "Create Volume" button
- "Attach Existing Volume" view
  - select an encrypted volume --> observe: reboot notice below radio buttons, disabled "Attach Volume" button
- Test the volume queries that were changed in this PR

## As an Author I have considered 🤔
